From: Jan Kara <jack@suse.cz>
Subject: bdi: Fix up kabi for dev_name addition
References: bsc#1171844
Patch-mainline: Never, kabi fixup

struct backing_dev_info is allocated by bdi_alloc_node() so it's safe to just
move added dev_name to the end and use __GENKSYMS__ trick.

Signed-off-by: Jan Kara <jack@suse.cz>

---
 include/linux/backing-dev-defs.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/backing-dev-defs.h
+++ b/include/linux/backing-dev-defs.h
@@ -220,7 +220,6 @@ struct backing_dev_info {
 	wait_queue_head_t wb_waitq;
 
 	struct device *dev;
-	char dev_name[64];
 	struct device *owner;
 
 	struct timer_list laptop_mode_wb_timer;
@@ -228,6 +227,9 @@ struct backing_dev_info {
 #ifdef CONFIG_DEBUG_FS
 	struct dentry *debug_dir;
 #endif
+#ifndef __GENKSYMS__
+	char dev_name[64];
+#endif
 };
 
 enum {
